lib/Hooks/ImageHooks.php

```diff
@@ -54,18 +54,88 @@ public function postCreate(Node $node) {
 		$logger = \OC::$server->getLogger();
 		$logger->log('debug', 'Image Path: '.$absolutePath, array('app' => 'MediaMetadata'));
 
+		$imageMetadata = new ImageMetadata();
+
 		if ($dimensions !== false) {
 			list($image_width, $image_height) = $dimensions;
 
 			$logger->log('debug', 'Image Height: '.$image_height, array('app' => 'MediaMetadata'));
 			$logger->log('debug', 'Image Width: '.$image_width, array('app' => 'MediaMetadata'));
 
-			$imageDimension = new ImageDimension();
-			$imageDimension->setImageId($node->getId());
-			$imageDimension->setImageHeight($image_height);
-			$imageDimension->setImageWidth($image_width);
+			$imageMetadata->setImageId($node->getId());
+			$imageMetadata->setImageHeight($image_height);
+			$imageMetadata->setImageWidth($image_width);
+		}
+
+		$exif = exif_read_data($absolutePath, 0, true);
+
+		if($exif !== false) {
+			$logger->info('EXIF Info', array('app' => 'MediaMetadata'));
+
+			foreach ($exif as $key => $section) {
+				foreach ($section as $name => $val) {
+					$logger->info($key . '.' . $name . ': ' . $val, array('app' => 'MediaMetadata'));
+				}
+			}
+
+			// Date Created
+			if((array_key_exists('EXIF', $exif)) AND (array_key_exists('DateTimeOriginal', $exif['EXIF']))) {
+				$date_created = $exif['EXIF']['DateTimeOriginal'];
+				$imageMetadata->setDateCreated($date_created);
+			}
 
-			$this->mapper->insert($imageDimension);
+			// GPS Latitude
+			if((array_key_exists('GPS', $exif)) AND (array_key_exists('GPSLatitude', $exif['GPS']))) {
+				$logger->notice('Keys of GPS Latitude: [0]- '.$exif['GPS']['GPSLatitude'][0].' [1]- '.$exif['GPS']['GPSLatitude'][1].' [2]- '.$exif['GPS']['GPSLatitude'][2],
```

Comment:
Please use debug as we don't want to pollute the logs with such information (I think?)

Also, instead of combining text and PHP, it's best to use placeholders, like in here: https://github.com/owncloud/gallery/blob/70de06213b69c03d9b9db231d611d996f4b8e086/controller/config.php#L49